nautobot_golden_config/models.py

```diff
@@ -177,32 +178,39 @@ def _verify_get_custom_compliance_data(compliance_details):
 
 def _get_hierconfig_remediation(obj):
     """Returns the remediating config."""
-    hierconfig_os = obj.device.platform.network_driver_mappings["hier_config"]
+    hierconfig_os = obj.device.platform.network_driver_mappings.get("hier_config")
+
     if not hierconfig_os:
-        raise ValidationError(f"platform {obj.network_driver} is not supported by hierconfig.")
+        raise ValidationError(f"platform {obj.device.platform.network_driver} is not supported by hierconfig.")
 
     try:
         remediation_setting_obj = RemediationSetting.objects.get(platform=obj.rule.platform)
     except Exception as err:  # pylint: disable=broad-except:
-        raise ValidationError(f"Platform {obj.network_driver} has no Remediation Settings defined.") from err
+        raise ValidationError(
+            f"Platform {obj.device.platform.network_driver} has no Remediation Settings defined."
```

Comment:
While testing this function, I noticed that `obj.network_driver` is not the correct syntax, it should be `obj.device.platform.network_driver`. This would cause a separate exception that is not handled.

Reply:
RemediationSetting is set per Platform, so probably we don't need network_driver here

Reply:
The network driver is specifically used to map to a hier config driver in netutils.